### PR TITLE
(6.3) update packages, use mongo for db, use blank page for redirect

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* Update project dependencies
* Update sample to use MongoDB (via Azure CosmosDB) instead of lowdb mocks
* Use blank page for redirect in SPA
* Minor refactoring

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/Azure-Samples/ms-identity-javascript-react-tutorial.git
cd ms-identity-javascript-react-tutorial
git checkout acrs-sample-update
cd 6-AdvancedScenarios/3-call-api-acrs/SPA
npm install
npm start
cd ../API
npm install
npm start
```

## What to Check
Verify that the following are valid
* Follow the README to run the sample E2E

### CODE REVIEW

**Comment by derisen** [comment]
keep open

File: 6-AdvancedScenarios/3-call-api-acrs/API/auth/routeGuard.js [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
 
-const authContextGuard = (req, res, next) => {
-    const acrs = AuthContext.getAuthContexts();
-    const isPut = req.method === "PUT";
-
-    
-    // if there is no auth context in the db, let the request through
+const authContextGuard = async (req, res, next) => {
+    const acrs = await AuthContext.getAuthContexts();
+    console.log(acrs);
```
Comment (salman90) [inline 6-AdvancedScenarios/3-call-api-acrs/API/auth/routeGuard.js:6]: Remove console.log

File: 6-AdvancedScenarios/3-call-api-acrs/README.md [hunk 2]
```diff
@@ -77,6 +78,18 @@ or download and extract the repository .zip file.
     npm install
```
Comment (salman90) [inline 6-AdvancedScenarios/3-call-api-acrs/README.md:78]: Add the path like in the API

**Review by salman90 — APPROVED** [review]
@derisen Looks good !

File: 6-AdvancedScenarios/3-call-api-acrs/API/auth/routeGuard.js (hunk 1 above)
Comment (derisen) [inline 6-AdvancedScenarios/3-call-api-acrs/API/auth/routeGuard.js:6]:
```suggestion
```

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/Configure.ps1 [hunk 3]
```diff
@@ -306,20 +306,20 @@ Function ConfigureApplications
 
     # Create the client AAD application
     Write-Host "Creating the AAD application (msal-react-spa-acrs)"
-    # create the application 
+    # create the application
     $clientAadApplication = New-AzureADApplication -DisplayName "msal-react-spa-acrs" `
         -HomePage "http://localhost:3000" `
-        -ReplyUrls "http://localhost:3000" `
+        -ReplyUrls "http://localhost:3000/redirect" `
```
Comment (derisen) [inline 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/Configure.ps1:312]:
```suggestion
        -ReplyUrls "http://localhost:3000/redirect.html" `
```

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
             "Kind": "SinglePageApplication",
             "Audience": "AzureADMyOrg",
             "HomePage": "http://localhost:3000",
-            "ReplyUrls": "http://localhost:3000",
+            "ReplyUrls": "http://localhost:3000/redirect",
```
Comment (derisen) [inline 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json:43]:
```suggestion
            "ReplyUrls": "http://localhost:3000, http://localhost:3000/redirect.html",
```

**Pull request merged**